packages/lib/src/components/ThreeDS2/components/DelegatedAuthentication/DelegatedAuthenticationEnrollment.tsx

```diff
@@ -0,0 +1,260 @@
+import { h, Fragment } from 'preact';
+import { useEffect, useState, useMemo } from 'preact/hooks';
+import {
+    DelegatedAuthenticationEnrollmentInputData,
+    DelegatedAuthenticationEnrollmentOutputData,
+    DelegatedAuthenticationEnrollmentProps,
+    DelegatedAuthenticationError
+} from './types';
+import { createChallengeResolveData, createOldChallengeResolveData, isInIframe } from '../utils';
+import { decodeBase64, encodeBase64 } from './utils';
+import Button from '../../../internal/Button';
+
+const DelegatedAuthenticationEnrollment = ({
+    token,
+    useOriginalFlow,
+    dataKey,
+    authorisationToken,
+    onComplete
+}: DelegatedAuthenticationEnrollmentProps) => {
+    const [status, setStatus] = useState<string>('loading');
+    const [paymentCredential, setPaymentCredential] = useState<PublicKeyCredential>(null);
+    const [registrationData, setRegistrationData] = useState<DelegatedAuthenticationEnrollmentInputData>(null);
+    const [supported, setSupported] = useState<boolean>();
+    const [spcSupported, setSpcSupported] = useState<boolean>();
+
+    const inIframe = useMemo(() => {
```

Comment:
This function doesn't need to memoized, it's not doing a heavy calculation just returning a boolean.